modules/setting/setting.go

```diff
@@ -415,7 +415,7 @@ func NewContext() {
 	case "duoshuo":
 		GravatarSource = "http://gravatar.duoshuo.com/avatar/"
 	case "gravatar":
-		GravatarSource = "//1.gravatar.com/avatar/"
+		GravatarSource = "//secure.gravatar.com/avatar/"
```

Comment:
Protocol-agnostic URLs aren't needed anymore. Just use `https://` unconditionally if it's available. See the note at the top of http://www.paulirish.com/2010/the-protocol-relative-url/ for more detail.